File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/UserScript.java
package io.flutter.plugins.webviewflutter;

import java.util.Map;

enum InjectionTime {
    DocumentStart, DocumentEnd;

    public static InjectionTime fromString(String str) {
        if (str != null && str.trim().toLowerCase() == "end") {
            return DocumentEnd;
        } else {
            return DocumentStart;
        }
    }
}

class UserScript {
    final String source;
    final InjectionTime injectionTime;
    final Boolean mainFrameOnly;

    UserScript(String src, InjectionTime injectionTime, Boolean mainFrame) {
        this.source = src;
        this.injectionTime = injectionTime != null ? injectionTime : InjectionTime.DocumentStart;
        this.mainFrameOnly = mainFrame != null ? mainFrame : false;            
    }

    UserScript(Map<String, Object> paramMap) {
        this(paramMap.get("source").toString(),
                InjectionTime.fromString((String) paramMap.get("injectionTime")),
                (Boolean) paramMap.get("mainFrameOnly"));
    }
}

File: packages/webview_flutter/android/src/main/kotlin/io/flutter/plugins/webviewflutter/UserScript.kt
package io.flutter.plugins.webviewflutter

internal enum class InjectionTime {
    DocumentStart, DocumentEnd;

    companion object {
        fun fromString(str: String?): InjectionTime {
            return if (str != null && str.trim().toLowerCase() === "end") {
                DocumentEnd
            } else {
                DocumentStart
            }
        }
    }
}

internal class UserScript constructor (val source: String, injectionTime: InjectionTime?, mainFrame: Boolean?) {
    val injectionTime: InjectionTime
    val mainFrameOnly: Boolean

    constructor(paramMap: Map<String, Any>) : this(
        paramMap["source"].toString(),
        InjectionTime.fromString(paramMap["injectionTime"] as String?),
        paramMap["mainFrameOnly"] as Boolean?
    ) {
    }

    init {
        this.injectionTime = injectionTime ?: InjectionTime.DocumentStart
        mainFrameOnly = mainFrame ?: false
    }
}
